### PR TITLE
Add io.github.xyproto.zsnes

### PR DESCRIPTION
**Pull request by JakobDev**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. xyproto/zsnes#10
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

`host:rw` is needed because it dosen't support Portals.

### CODE REVIEW

**Comment by JakobDev** [comment]
bot, build io.github.xyproto.zsnes

File: io.github.xyproto.zsnes.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  <!--Created with jdAppStreamEdit 6.0-->
+  <id>io.github.xyproto.zsnes</id>
+  <name>ZSNES</name>
+  <summary>Super Nintendo emulator</summary>
+  <developer_name>Alexander F. Rødseth</developer_name>
+  <launchable type="desktop-id">io.github.xyproto.zsnes.desktop</launchable>
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>GPL-2.0</project_license>
+  <description>
+    <p>ZSNES is a classical SNES Emulator. It's development started 1998 and is was one of the most uses SNES Emulators for years.</p>
```
Comment (hfiguiere) [inline io.github.xyproto.zsnes.metainfo.xml:12]:
```suggestion
    <p>ZSNES is a classical SNES Emulator. Its development started 1998 and is was one of the most uses SNES Emulators for years.</p>
```

**Comment by hfiguiere** [comment]
bot, build io.github.xyproto.zsnes

**Review by hfiguiere — CHANGES_REQUESTED** [review]
I don't see the files being upstream.

Please make sure to submit a PR.

File: io.github.xyproto.zsnes.yaml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+runtime-version: '22.08'
+sdk: org.freedesktop.Sdk
+command: zsnes
+finish-args:
+  - --share=ipc
+  - --device=all
+  - --socket=x11
+  - --allow=multiarch
+  - --socket=pulseaudio
+  - --filesystem=host:rw
```
Comment (hfiguiere) [inline io.github.xyproto.zsnes.yaml:12]: this remain a very broad permission

Comment (JakobDev) [reply io.github.xyproto.zsnes.yaml:12]: It#s the only Permission that makes it fully work. And is it not more harmful than `home`.

File: io.github.xyproto.zsnes.yaml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  libdir: /app/lib32
+
+modules:
+  - shared-modules/SDL/SDL-1.2.15.json
+
+  - name: zsnes
+    buildsystem: simple
+    build-commands:
+      - make
+      - mkdir -p /app/lib/i386-linux-gnu
```
[the remark this answers is not in the document]
Comment (JakobDev) [reply io.github.xyproto.zsnes.yaml:53]: It contains lot (46%) of Assembler code, so this is will not work for 64bit or for ARM.

**Comment by JakobDev** [comment]
> I don't see the files being upstream.

Everything is included in the linked Issue.

**Pull request closed without merging**